lib/Doctrine/ORM/Mapping/ClassMetadata.php

```diff
@@ -715,8 +720,20 @@ public function isRootEntity()
     public function isInheritedProperty($fieldName)
     {
         $declaringClass = $this->properties[$fieldName]->getDeclaringClass();
+        if ($declaringClass->className === $this->className) {
+            return false;
+        } else if (! $declaringClass->isMappedSuperclass) {
+            return true;
+        }
+
+        $parentClass = $this;
+        while (($parentClass = $parentClass->getParent()) !== $declaringClass) {
```

Comment:
You override `$parentClass` here while above you set it to `$this`. Should it be `$this->getParent()` to be more readable or is there a reason for this choice? Something similar can be found in `ClassMetadata::getRootClassName()`.

Reply:
This is just a means of iterating up the hierarchy. The same method is used a number of times already in JoinedSubclassPersister.php.